Xamarin.Essentials/Sms/Sms.shared.cs

```diff
@@ -31,6 +31,11 @@ public SmsMessage()
         {
         }
 
+        public SmsMessage(string body, string recipient)
```

Comment:
May I suggest outsourcing the body setting to a private constructor? this way we can elide setting the recipient list by allocating an array and copy constructing a list as well as allocating linq decorators. 

```
    public SmsMessage(string body, string recipient)
            : this(body)
        {
            Recipients = string.IsNullOrWhiteSpace(body) ? new List<string>(0) : new List<string>() {recipient};
        }

        private SmsMessage(string body)
        {
            Body = body;
        }

        public SmsMessage(string body, IEnumerable<string> recipients)
         : this(body)
        {
            Recipients = recipients?.Where(x => !string.IsNullOrWhiteSpace(x)).ToList();
        }`

Reply:
Good point. I would even say go further and forget about the chaining of constructors:

```csharp
public SmsMessage() {
}
public SmsMessage(string body, string recipient) {
    Body = body;
    if (!string.IsNullOrWhiteSpace(recipient))
        Recipients.Add(recipient);
}
public SmsMessage(string body, IEnumerable<string> recipients) {
    Body = body;
    if (recipients != null) {
        Recipients.AddRange(recipients.Where(x => !string.IsNullOrWhiteSpace(x)));
    }
}
// . . .
public List<string> Recipients { get; set; } = new List<string>();
```

We don't have to care about the initial size of the list as the default size is 4 (if corefx is anything to go by).